### PR TITLE
H-5198: Implement PropertyType generation and persistence MVP with known limitations

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

This PR implements a **PropertyType seeding system** for benchmark scenarios, enabling deterministic generation and persistence of PropertyType datasets. The implementation follows the established seeding architecture patterns from DataType and User seeding, providing a foundation for large-scale benchmark testing.

While functional for current benchmark needs, the implementation includes acknowledged performance limitations and missing features that are tracked separately for future optimization.

### Core Implementation:

- **PropertyType Producer System** (`tests/graph/test-data/rust/src/seeding/producer/property_type.rs`)
  - Deterministic PropertyType generation with configurable parameters
  - Support for Local and Remote ownership types
  - URL construction with proper BaseUrl handling and error propagation
  - Two-phase approach: Configuration → Binding (with catalogs) → Distribution sampling

- **PropertyValues Distribution System** (`tests/graph/test-data/rust/src/seeding/distributions/ontology/property_type/values.rs`)
  - `DataTypeCatalog` trait for PropertyType value generation
  - `InMemoryDataTypeCatalog` implementation with proper error handling
  - Safe array access using `SliceRandom::choose()` instead of manual indexing
  - Empty catalog validation and comprehensive error propagation

- **Benchmark Integration** (`tests/graph/benches/graph/scenario/stages/property_type.rs`)
  - `GeneratePropertyTypesStage` - PropertyType generation from configs
  - `PersistPropertyTypesStage` - Database persistence with Local/Remote handling
  - `BuildDataTypeCatalogStage` - DataType catalog construction for PropertyType dependencies
  - Proper error handling for edge cases (empty catalogs, missing dependencies)

### Configuration & Infrastructure:

- **Configuration System** (`tests/graph/benches/graph/config/mod.rs`)
  - `PROPERTY_TYPE_PRODUCER_CONFIGS` LazyLock for config loading
  - Integration with existing config loading patterns

- **Scenario Updates** (`tests/graph/benches/config/scenarios/full_ontology_seeding.json`)
  - Complete PropertyType pipeline: Generation → Catalog Building → Persistence
  - Proper stage ordering to avoid resource consumption issues

- **Resource Management** (`tests/graph/benches/graph/scenario/runner.rs`)
  - Extended `Resources` struct with `property_types` and `data_type_catalogs`
  - Thread-safe resource sharing between stages

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

- **Performance**: PropertyType persistence currently uses individual database inserts (~20ms per PropertyType). Bulk insert optimization is tracked in H-5204.
- **Feature Completeness**: Array and Object PropertyValue types are not yet implemented (TODOs in place).
- **Memory Usage**: Current implementation loads all PropertyTypes into memory before persistence. Streaming implementation needed for very large datasets.
- **Manual Testing**: Full integration tests with large datasets (50M+ PropertyTypes) need to be run manually due to resource constraints.

## 🐾 Next steps

- **Entity Type Seeding**: Extend the seeding system to support EntityTypes following the same patterns
- **Advanced PropertyValue Types**: Implement Array and Object PropertyValue generation (H-TBD)
- **Streaming Implementation**: Add memory-efficient streaming for very large datasets

## 🛡 What tests cover this?

- **Unit Tests**: Comprehensive test coverage for all distribution configurations and edge cases
- **Integration Tests**: PropertyType generation, binding, and sampling workflows
- **Benchmark Tests**: Full pipeline testing from generation through database persistence
- **Error Handling Tests**: Empty catalog validation, malformed configuration handling

## ❓ How to test this?

1. **Run PropertyType Generation Test**:

   ```bash
   cd tests/graph/benches
   cargo test --bench graph -- --nocapture
   ```

2. **Verify Performance** (Release mode for accurate metrics):

   ```bash
   cargo test --release --bench graph -- --nocapture
   ```

3. **Check Generated Data**: Confirm PropertyTypes are properly persisted in the database with correct Local/Remote ownership types

4. **Validate Error Handling**: Test with empty DataType catalogs to ensure proper error propagation

## 📹 Demo

**Performance Results (Release Mode):**

- **Generation**: 50M PropertyTypes in ~22 seconds (~2.25M PropertyTypes/second)
- **Persistence**: 500 PropertyTypes in ~10 seconds (limited by individual DB inserts)
- **Memory Usage**: Efficient for moderate datasets, streaming needed for 100M+ scale
- **Deterministic**: Reproducible results across runs with same seed

**Current Capabilities:**

- Successfully tested up to 50M PropertyTypes generation
- Architecture designed for 100M+ PropertyType benchmarks with future optimizations
- Clean separation of concerns enabling incremental performance improvements
- Foundation established for Entity Type seeding implementation

### CODE REVIEW

File: tests/graph/test-data/rust/src/seeding/producer/ontology/domain.rs
Comment (TimDiekmann) [inline tests/graph/test-data/rust/src/seeding/producer/ontology/domain.rs:1]: This was mainly moved from `data_type.rs` to be re-used for property types

File: tests/graph/benches/graph/scenario/stages/property_type.rs [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+            org_catalog: None::<&InMemoryWebCatalog>,
+            data_type_catalog,
+        };
+
+        let stage_id = self
+            .stage_id
+            .map_or_else(|| StageId::from_name(&self.id), StageId::new);
+
+        // Generate all params first to avoid borrow conflicts
+        // TODO: implement streaming to avoid loading all property types into memory at once for
```
Comment (indietyp) [inline tests/graph/benches/graph/scenario/stages/property_type.rs:111]: add Linear ticket if applicable

File: tests/graph/test-data/rust/src/seeding/distributions/ontology/property_type/values.rs [hunk 2]
```diff
@@ -0,0 +1,543 @@
+//! Distributions for generating `PropertyValues` structures.
+//!
+//! This module provides configurable distributions for generating [`PropertyValues`] that can
+//! represent different value structures:
+//!
+//! - Simple data type references (currently implemented)
+//! - Array structures (TODO: not yet implemented)
```
Comment (indietyp) [inline tests/graph/test-data/rust/src/seeding/distributions/ontology/property_type/values.rs:7]: Linear ticket would be appreciated
